WPFHospitalEditor/AdditionalInformation.xaml.cs

```diff
@@ -77,20 +76,39 @@ private void Done_Click(object sender, RoutedEventArgs e)
 
         private void refreshMap()
         {
-            AllMapObjects allMapObjects = new AllMapObjects();
+            String buildingId = building.getBuildingId(mapObject);
+            List<MapObject> allMapObjects = mapObjectController.getAllMapObjects();
+            List<MapObject> objectsForRefreshing = new List<MapObject>();
+            foreach(MapObject mapObjectIterate in allMapObjects)
+            {
+                if (!mapObjectIterate.Description.Equals(""))
+                {
+                    compare(objectsForRefreshing, mapObjectIterate, buildingId);                   
+                }
+            }
             building.canvas.Children.Clear();
-            building.hospitalMap.addObjectToCanvas(building.buildingFloors[floor].getAllFloorMapObjects(), building.canvas);
+            building.hospitalMap.addObjectToCanvas(objectsForRefreshing, building.canvas);
+        }
+
+        private void compare(List<MapObject> objectsForRefreshing, MapObject mapObjectIterate, String buildingId)
+        {
+            
+            String[] firstSplit = mapObjectIterate.Description.Split("&");
+            String[] secondSplit = firstSplit[0].Split("-");
+            if (secondSplit[0].Equals(buildingId) && secondSplit[1].Equals(floor.ToString()))
+            {
+                objectsForRefreshing.Add(mapObjectIterate);
+            }
```

Comment:
if the name of method is compare then you should do just that and return boolean. This method has side effects

Reply:
I changed refreshMap method and this method no longer exists.